### PR TITLE
Improve testing around search result IDs.

### PR DESCRIPTION
**Pull request by kadams54**
The change came about because we had a bug (fixed in https://github.com/juju/juju-gui/pull/1658) that the tests missed because they didn't use particularly realistic IDs. Consequently code for parsing the IDs was never being exercised.

In the course of improving the tests, I found another bug and fixed that as well.

### CODE REVIEW

**Comment by kadams54** [comment]
## QA

1. Search for "sql" and filter the search to just bundles.
2. Ensure that you can click through to details AND deploy both a promulgated bundle and a non-promulgated one, ideally in a real(ish) env (i.e., not sandbox).

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5074/
Test FAILed.

**Comment by bac** [comment]
👍

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5075/
Test PASSed.

File: jujugui/static/gui/src/app/components/search-results/search-results.js [hunk 1]
```diff
@@ -138,7 +138,8 @@ YUI.add('search-results', function(Y) {
             storeId = obj.storeId;
         if (!storeId || storeId.indexOf(ownerPath) < 0) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/search-results/search-results.js]: There is a lot of logic here that makes no sense to the casual reader. Could you add a code comment here outlining why we have to do these things.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5076/
Test FAILed.

**Comment by bac** [comment]
QA was good.  On GI Joe running on lxd I deployed the Adam Stokes non-promulgated bundle that shows up with a 'sql' search. Four machines came up fine. Then I deployed the small promulgated bundle from the same search. Those five machines came up. 

Looks good.  Thank you.

**Comment by kadams54** [comment]
Thanks all! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5080/
Test PASSed.

**Pull request merged**